### PR TITLE
Better job log: Trying to fix the frontend mutations for AdminJobLogPage and JobsTable

### PR DESCRIPTION
**Pull request by vincentc0202**
This is to close [#3]. This feature is to improve the functionality and appearance of the Job Logs page for Admins. 

Dokku deployment:
https://better-logs.dokku-01.cs.ucsb.edu/

### CODE REVIEW

**Review by cameronbyrne4 — DISMISSED** [review]
LGTM. Lots of tests addded!

File: frontend/src/main/components/Jobs/JobsTable.js [hunk 1]
```diff
@@ -1,8 +1,6 @@
 import React from "react";
-import OurTable, {
-  PlaintextColumn,
-  DateColumn,
-} from "main/components/OurTable";
+import OurTable, { DateColumn } from "main/components/OurTable";
+import { Link } from "react-router-dom";
```
Comment (pconrad) [inline frontend/src/main/components/Jobs/JobsTable.js:3]: To support the suggestion later in the code review...

```suggestion
import { Link } from "react-router-dom";
import Plaintext from "main/components/Utils/Plaintext";
```

File: frontend/src/main/components/Jobs/JobsTable.js [hunk 2]
```diff
@@ -18,7 +16,38 @@ export default function JobsTable({ jobs }) {
...
+            </div>
+          );
+        }
+        const logLines = log.split("\n");
+        const truncatedLog = logLines.slice(0, 10).join("\n");
+        return (
+          <div data-testid={`JobsTable-cell-row-${cell.row.index}-col-Log`}>
+            {logLines.length > 10 ? (
+              <>
+                <pre>{truncatedLog}</pre>
```
Comment (pconrad) [inline frontend/src/main/components/Jobs/JobsTable.js]: Try this instead, and see if you get a nicer looking result:
```suggestion
              <Plaintext text={truncatedLog} />,
```

File: frontend/src/main/components/Jobs/JobsTable.js [hunk 3]
```diff
@@ -1,8 +1,6 @@
 import React from "react";
-import OurTable, {
-  PlaintextColumn,
-  DateColumn,
-} from "main/components/OurTable";
+import OurTable, { DateColumn } from "main/components/OurTable";
```
Comment (pconrad) [inline frontend/src/main/components/Jobs/JobsTable.js:2]: Goes with later suggestion

```suggestion
import OurTable, { DateColumn } from "main/components/OurTable";
import BasicLayout from "main/layouts/BasicLayout/BasicLayout";
```

File: frontend/src/main/pages/Admin/AdminJobLogPage.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    method: "GET",
+    url: `/api/jobs/logs/${id}`,
+  });
+
+  const handleBack = () => {
+    navigate("/admin/jobs");
+  };
+
+  return (
+    <div>
```
Comment (pconrad) [inline frontend/src/main/pages/Admin/AdminJobLogPage.js]: Look at every single other Page component in the entire app.  What do you notice?

```suggestion
    <BasicLayout>
```

File: frontend/src/main/pages/Admin/AdminJobLogPage.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      <Button onClick={handleBack}>Back to Job List</Button>
+      <h3>Job Logs for Job {id}</h3>
+      {jobLogs ? <pre>{jobLogs}</pre> : <p>Loading...</p>}
+    </div>
```
Comment (pconrad) [inline frontend/src/main/pages/Admin/AdminJobLogPage.js]:
```suggestion
    </BasicLayout>
```

**Review by pconrad — CHANGES_REQUESTED** [review]
It's important to get to know the code base, and look at how things are implemented.

For example *every* page component uses `<BasicLayout>`.  Is there a good reason for this new page to be any different?

And: there's already a solution to getting the logs to show up in a fixed width font and one that respects line breaks.  Use it.

File: frontend/src/main/components/Jobs/JobsTable.js (hunk 3 above)
Comment (vincentc0202) [reply frontend/src/main/components/Jobs/JobsTable.js:2]: I understand why AdminJobLogPage needs BasicLayout, but does JobsTable need it as well? I checked the other files within the same folder as this file and none of them use BasicLayout.

Comment (pconrad) [reply frontend/src/main/components/Jobs/JobsTable.js:2]: No it doesn't... may have put this suggestion in the wrong file.  Sorry about that.

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**